src/extension/alterschema/linter/then_false.h

```diff
@@ -0,0 +1,52 @@
+class ThenFalse final : public SchemaTransformRule {
+public:
+  ThenFalse()
+      : SchemaTransformRule{
+            "then_false",
+            "`if: S, then: false` is logically equivalent to `not: S`"} {};
+
+  [[nodiscard]] auto
+  condition(const JSON &schema, const JSON &, const Vocabularies &vocabularies,
+            const SchemaFrame &, const SchemaFrame::Location &,
+            const SchemaWalker &, const SchemaResolver &) const
+      -> SchemaTransformRule::Result override {
+    if (!contains_any(vocabularies,
```

Comment:
I think this whole condition could be a single chained `&&` line, like some of the other rules. I think it makes it easier to read, otherwise we have to read the inverse conditionals in the whole body. For the `for` loop, you can use `std::all_of`, `std::any_of`, or whatever